components/navigation/ProfileMenuMemberships.tsx

```diff
@@ -111,63 +111,48 @@ const MembershipLine = ({ user, membership, closeDrawer }) => {
 
 MembershipLine.propTypes = {
   user: PropTypes.object,
-  membership: PropTypes.object,
+  collective: PropTypes.object,
   closeDrawer: PropTypes.func,
 };
 
-const sortMemberships = (memberships: LoggedInUser['memberOf']) => {
-  if (!memberships?.length) {
+const sortCollectives = (collectives: LoggedInUser['memberOf'][0]['collective'][]) => {
+  if (!collectives?.length) {
     return [];
   } else {
-    return memberships.sort((a, b) => {
-      return a.collective.slug.localeCompare(b.collective.slug);
+    return collectives.sort((a, b) => {
+      return a.slug.localeCompare(b.slug);
     });
   }
 };
 
-const filterArchivedMemberships = (memberships: LoggedInUser['memberOf']) => {
-  const archivedMemberships = memberships.filter(m => {
-    if (
-      m.role !== 'BACKER' &&
-      m.collective.isArchived &&
-      !(m.collective.type === 'EVENT' && isPastEvent(m.collective))
-    ) {
-      return true;
-    } else {
-      return false;
-    }
-  });
+const getAllCollectivesForMenu = (memberships: LoggedInUser['memberOf']) => {
+  const collectives = memberships.filter(m => m.role !== 'BACKER' && m.collective).map(m => m.collective);
+  const administratedChildren = memberships
+    .filter(m => m.role === 'ADMIN' && m.collective?.children?.length)
+    .map(m => m.collective.children.map(c => ({ ...c, parentCollective: m.collective })))
+    .flat();
```

Comment:
I think we should limit the number of Events we display on these menus, as it can be overwhelming for some users:
![image](https://github.com/opencollective/opencollective-frontend/assets/2119706/297daa99-b40c-4c9e-92d2-fa4794b698d0)

Reply:
We normally don't display past events in there, I'll double check